pkg/deployer/deployer.go

```diff
@@ -0,0 +1,91 @@
+package deployer
```

Comment:
please add tests for this. they don't have to be comprehensive, just basic tests that will let us add new tests in future as bugs come up

Reply:
add the license to this file

Reply:
Can I open an issue to add tests and address it later? This PR is already kinda big